### PR TITLE
419 Show Calendar - add tooltip calendar

### LINKED ISSUE #419: Show Calendar - Tooltip calendar
## Description of Issue

Add the tooltip calendar that appears in the upper left corner of the wireframes at https://drive.google.com/file/d/13xQLl3P91aNjqmIs1vdUzmtxQkesDgDU/view?usp=sharing (page 2)

## Possible Solution and/or Approach

Look at https://github.com/airbnb/react-dates as a possible solution

### PR DESCRIPTION
**Pull request by seankwilliams**
## Guidelines

DEVELOPMENT branch on the LEFT <br>
YOUR branch on the RIGHT

## Issue

This issue resolves #419 

## Description

Adds a small tooltip calendar to the left corner of the Show Calendar page.

### CODE REVIEW

File: client/src/components/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -23,13 +23,13 @@ class Dropdown extends Component {
     let button = [];
     switch (type) {
       case 'icon':
-        button.push(<DropdownIcon icon={this.props.icon} />);
+        button.push(<DropdownIcon key="icon" icon={this.props.icon} />);
```
Comment (seankwilliams) [inline client/src/components/Dropdown/Dropdown.js]: this, and the other change in this file, are to stop the JavaScript error that appears asking for `key` to be provided in this component. The change is just to eliminate the error, this doesn't change any functionality.

File: client/src/components/Shows/ShowCalendar.js [hunk 2]
```diff
@@ -109,24 +115,29 @@ class Calendar extends Component {
...
 
   render() {
-    const { shows } = this.props;
+    const { date, shows } = this.props;
     const localizer = BigCalendar.momentLocalizer(moment);
 
+    //if date provided in properties, always have the calendar display that date
+    //otherwise, just start with today
+    let calendarDateProperty =
+      typeof date == 'undefined' ? { defaultDate: new Date() } : { date: date };
```
Comment (jberns) [inline client/src/components/Shows/ShowCalendar.js]: I think defaultDate can always stay on the calendar and then this is overwritten by the date when you pass it.  I had problems with changing dates in the past without defaultDate so I'm not sure if leaving it off makes a difference once we apply the real date.

File: client/src/pages/CalendarHomePage/CalendarHomePage.js [hunk 3]
```diff
@@ -18,24 +21,59 @@ class CalendarHomePage extends Component {
...
+          <DayPickerSingleDateController
+            key={dayPickerKey}
+            date={selectedDate}
+            horizontalMonthPadding={0}
+            noBorder={true}
+            daySize={28}
+            numberOfMonths={1}
+            onDateChange={this.dateChangeHandler}
+            focused={true}
+            onFocusChange={({ focused }) => this.setState({ focused: true })} // Force the focused states to always be truthy so that date is always selectable
```
Comment (jberns) [inline client/src/pages/CalendarHomePage/CalendarHomePage.js]: Component looks good!  We can remove the shortcut menu with ```hideKeyboardShortcutsPanel={true}``` if you want the home page to look a bit cleaner.

**Comment by seankwilliams** [comment]
thanks @jberns ! I implemented both pieces of feedback.

**Pull request merged**